.changeset/stupid-lemons-rest.md

```diff
@@ -0,0 +1,5 @@
+---
+"@apollo/query-planner": patch
+---
+
+Fix issue with missing fragment definitions due to generateQueryFragments
```

Comment:
I can't send this as a GH suggestion due to use of triple backtick in the comment, but would you mind adding this or something like it?

> Fix issue with missing fragment definitions due to generateQueryFragments
> 
> An incorrect impementation detail in `generateQueryFragments` caused certain queries to be missing fragment definitions. Specifically, _subsequent_ fragment "candidates" with the same type condition and the same length of selections as a previous fragment weren't correctly added to the list of fragments. An example of an affected query is:
> 
> ```graphql
> query {
>   t {
>     ... on A {
>       x
>       y
>     }
>   }
>   t2 {
>     ... on A {
>       y
>       z
>     }
>   }
> }
> ```
> 
> In this case, the second selection set would be converted to an inline fragment spread to subgraph fetches, but the fragment definition would be missing.

Reply:
Ideally the changeset should have the ```graphql code fencing around the query, but that doesn't paste over from my comment.